### PR TITLE
Add leave button and minor refactoring

### PR DESCRIPTION
**Pull request by herariom**
# What
* Add leave CTF slash command
* Add leave button for CTFs, posted as the first message in a newly created CTF channel
# Why 
So that users can leave CTFs
# How
Add new slash command and callback for leaving CTFs, similar to the functionality to join CTFs
# Testing
Tested to ensure proper functionality of leaving and no regressions in functionality for creating and joining CTFs

### CODE REVIEW

File: commands.go [hunk 1]
```diff
@@ -106,11 +130,23 @@ func ctfCommandCallback(s *discordgo.Session, i *discordgo.InteractionCreate) {
...
+			},
+		}
+		leaveRespComponent = append(leaveRespComponent, leaveActionRow)
+
 		// Create a channel for the CTF that is locked to the created role
 		targetGuild, _ := s.Guild(i.GuildID)
 		everyoneID := targetGuild.Roles[0].ID // Per my assumption, @everyone is the first role - Jack
 
-		s.GuildChannelCreateComplex(i.GuildID, discordgo.GuildChannelCreateData{
+		channel, _ := s.GuildChannelCreateComplex(i.GuildID, discordgo.GuildChannelCreateData{
```
Comment (weakphish) [inline commands.go]: I would add error handling here if possible, assuming that the second value is an error. If an error is detected, lets log it with `log.Println` and maybe return a message to the user in channel (ephemeral?)

File: commands.go [hunk 2]
```diff
@@ -130,67 +166,104 @@ func ctfCommandCallback(s *discordgo.Session, i *discordgo.InteractionCreate) {
...
+	})
+	if err != nil {
+		log.Print(err)
+	}
+}
+
+func joinCTF(s *discordgo.Session, i *discordgo.InteractionCreate) {
+
+	ctfName := i.Message.Content
+	joinCTFCallback(s, i, ctfName)
```
Comment (weakphish) [inline commands.go:219]: There might be a cleaner way to do this without two chained functions. Thoughts?

File: commands.go [hunk 3]
```diff
@@ -130,67 +166,104 @@ func ctfCommandCallback(s *discordgo.Session, i *discordgo.InteractionCreate) {
...
+}
 
-		// Reply with success
+func leaveCTF(s *discordgo.Session, i *discordgo.InteractionCreate) {
+	ctfName := i.Message.Content
+	leaveCTFCallback(s, i, ctfName)
```
Comment (weakphish) [inline commands.go:224]: Same as previous comment in joinCTF

File: commands.go [hunk 4]
```diff
@@ -130,67 +166,104 @@ func ctfCommandCallback(s *discordgo.Session, i *discordgo.InteractionCreate) {
...
+
+func joinCTFCallback(s *discordgo.Session, i *discordgo.InteractionCreate, ctfName string) {
+	guild, err := s.Guild(i.GuildID)
+	if err != nil {
 		s.InteractionRespond(i.Interaction, &discordgo.InteractionResponse{
 			Type: discordgo.InteractionResponseChannelMessageWithSource,
 			Data: &discordgo.InteractionResponseData{
 				Flags:   1 << 6,
-				Content: "Added user " + callingUser.Username + " to role " + ctfRole.Name,
+				Content: "Couldn't find the calling Guild",
```
Comment (weakphish) [inline commands.go:234]: Add logging with `log.Println`

File: commands.go (hunk 1 above)
Comment (herariom) [reply commands.go]: Added

File: commands.go (hunk 4 above)
Comment (herariom) [reply commands.go:234]: Added

File: commands.go (hunk 2 above)
Comment (herariom) [reply commands.go:219]: Resolved in discussion

**Pull request merged**